nio/src/main/scala/zio/nio/file/Files.scala

```diff
@@ -82,13 +96,26 @@ object Files {
     effectBlocking(Path.fromJava(JFiles.createTempDirectory(dir.javaPath, prefix.orNull, fileAttributes.toSeq: _*)))
       .refineToOrDie[IOException]
 
+  def createTempDirectoryManaged(
+    dir: Path,
+    prefix: Option[String],
+    fileAttributes: Iterable[FileAttribute[_]]
+  ): ZManaged[Blocking, IOException, Path] =
+    ZManaged.make(createTempDirectory(dir, prefix, fileAttributes))(release = deleteRecursive(_).ignore)
+
   def createTempDirectory(
     prefix: Option[String],
     fileAttributes: Iterable[FileAttribute[_]]
   ): ZIO[Blocking, IOException, Path] =
     effectBlocking(Path.fromJava(JFiles.createTempDirectory(prefix.orNull, fileAttributes.toSeq: _*)))
       .refineToOrDie[IOException]
 
+  def createTempDirectoryManaged(
+    prefix: Option[String],
+    fileAttributes: Iterable[FileAttribute[_]]
+  ): ZManaged[Blocking, IOException, Path] =
+    ZManaged.make(createTempDirectory(prefix, fileAttributes))(release = deleteRecursive(_).ignore)
```

Comment:
Not 100% sure about recursive directory deletion. But it seems to be reasonable in terms of temporary directories